src/lib.rs

```diff
@@ -1,15 +1,220 @@
+#[cfg(all(feature = "no-std", not(test)))]
+#[no_std]
+
+#[cfg(any(feature = "default", feature = "std", test))]
+#[macro_use]
+extern crate std;
+
+#[cfg(all(feature = "no-std", not(test)))]
+#[macro_use]
+extern crate core as std;
+
+#[cfg(all(feature = "no-std", not(test)))]
+extern crate alloc;
+
+#[cfg(all(feature = "no-std", not(test)))]
+use alloc::vec::Vec;
+
+#[cfg(all(feature = "no-std", not(test)))]
+use alloc::collections::btree_map::BTreeMap;
+#[cfg(any(feature = "default", feature = "std", test))]
+use std::collections::BTreeMap;
+
 extern crate serde;
+#[cfg(any(feature = "default", feature = "std", test))]
 extern crate serde_json;
+#[cfg(all(feature = "no-std", not(test)))]
+extern crate serde_json_core;
 
-use serde::{Serialize, Deserialize};
+use serde::{Deserialize, Serialize};
+
+//Rust-based transcription of Package standard as specified at https://github.com/ethpm/ethpm-spec/blob/master/spec/package.spec.json
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct Package<'a> {
+    manifest_version: &'a str, //TODO: Rustify
+    package_name: &'a str,
+    meta: Option<PackageMeta<'a>>,
+    version: &'a str, //TODO: Rustify
+    sources: Option<BTreeMap<&'a str, &'a str>>, //TODO: Rustify
+    contract_types: Option<BTreeMap<&'a str, ContractType<'a>>>, //TODO: Rustify
+    deployments: Option<BTreeMap<&'a str, BTreeMap<&'a str, ContractInstance<'a>>>>, //TODO: Rustify
+    build_dependencies: Option<BTreeMap<&'a str, &'a str>>, //TODO: Rustify
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct PackageMeta<'a> {
+    authors: Option<Vec<&'a str>>,
+    license: Option<&'a str>,
+    description: Option<&'a str>,
+    keywords: Option<Vec<&'a str>>,
+    links: Option<BTreeMap<&'a str, &'a str>>, //TODO: Rustify
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ContractType<'a> {
+    contract_name: Option<&'a str>,
+    deployment_bytecode: Option<BytecodeObject<'a>>,
+    runtime_bytecode: Option<BytecodeObject<'a>>,
+    abi: Option<Vec<ABI<'a>>>,
+    natspec: Option<NatSpec<'a>>,
+    compiler: Option<CompilerInformation<'a>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ContractInstance<'a> {
+    contract_type: &'a str,
+    address: &'a str,
+    transaction: Option<&'a str>,
+    block: Option<&'a str>,
+    deployment_bytecode: Option<BytecodeObject<'a>>,
+    runtime_bytecode: Option<BytecodeObject<'a>>,
+    compiler: Option<CompilerInformation<'a>>,
+    link_dependencies: Option<Vec<LinkValue<'a>>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct BytecodeObject<'a> {
+    bytecode: Option<&'a str>,
+    link_references: Option<Vec<LinkReference<'a>>>,
+    link_dependencies: Option<Vec<LinkValue<'a>>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ABI<'a> {
```

Comment:
Might be helpful to leverage this library for Ethereum ABIs: https://docs.rs/ethabi/8.0.1/ethabi/

Reply:
Tracked in #2